src/Altinn.Common/Altinn.Common.AccessToken/Altinn.Common.AccessToken/AccessTokenHandler.cs

```diff
@@ -154,5 +157,19 @@ private StringValues GetAccessTokens()
 
             return StringValues.Empty;
         }
+
+        private void SetAccessTokenCredential(string issuer, ClaimsPrincipal claimsPrincipal)
+        {
+            string appClaim = string.Empty;
+            foreach (Claim claim in claimsPrincipal.Claims)
+            {
+                if (claim.Type.Equals(AccessTokenClaimTypes.App))
+                {
+                    appClaim = claim.Value;
```

Comment:
you could break the loop here when the claim is found.

Reply:
done